manager/volume.go

```diff
@@ -633,7 +633,7 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 
 	kubernetesStatus := &v.Status.KubernetesStatus
 	if kubernetesStatus.PVCName != "" && kubernetesStatus.LastPVCRefAt == "" {
-		pvc, err := m.ds.GetPersistentVolumeClaim(kubernetesStatus.Namespace, kubernetesStatus.PVCName)
+		pvc, err := m.ds.GetPersistentVolumeClaimRO(kubernetesStatus.Namespace, kubernetesStatus.PVCName)
```

Comment:
Since you are modifying the PVC you want to use `GetPersistentVolumeClaim` then return a deep copy in that non ReadOnly function call.

While you are fixing this also return a deep copy from the `GetPersistentVolume` which is used by the PVC creation / cleanup, which again is modifying the raw instance.

Reply:
changed